cime_config/machines/cmake_macros/crayclanggpu_crusher.cmake

```diff
@@ -0,0 +1,52 @@
+if (compile_threaded)
+  string(APPEND FFLAGS   " -fopenmp")
+  string(APPEND CFLAGS   " -fopenmp")
+  string(APPEND CXXFLAGS " -fopenmp")
+  string(APPEND LDFLAGS  " -fopenmp")
+endif()
+if (DEBUG)
+  string(APPEND CFLAGS   " -O0 -g")
+  string(APPEND FFLAGS   " -O0 -g")
+  string(APPEND CXXFLAGS " -O0 -g")
+  string(APPEND CPPDEFS " -DYAKL_DEBUG")
+endif()
+string(APPEND CPPDEFS " -DFORTRANUNDERSCORE -DNO_R16 -DCPRCRAY")
+string(APPEND FC_AUTO_R8 " -s real64")
+string(APPEND FFLAGS " -f free -N 255 -h byteswapio -em")
+if (NOT compile_threaded)
+  string(APPEND FFLAGS " -M1077")
+endif()
+string(APPEND FFLAGS_NOOPT " -O0")
+set(HAS_F2008_CONTIGUOUS "TRUE")
+string(APPEND LDFLAGS " -Wl,--allow-multiple-definition")
+set(SUPPORTS_CXX "TRUE")
+set(CXX_LINKER "FORTRAN")
+set(MPICC "cc")
+set(MPICXX "hipcc")
+set(MPIFC "ftn")
+set(SCC "cc")
+set(SCXX "hipcc")
+set(SFC "ftn")
+
+if (NOT DEBUG)
+  string(APPEND CFLAGS   " -O2")
+  string(APPEND CXXFLAGS " -O2")
+  string(APPEND FFLAGS   " -O2")
+endif()
+
+if (COMP_NAME STREQUAL elm)
+  string(APPEND FFLAGS " -hfp0")
+endif()
+string(APPEND FFLAGS " -hipa0 -hzero -em -ef -hnoacc")
```

Comment:
Please, remove -hzero as per comment https://github.com/E3SM-Project/scream/pull/2129/files#r1072914312 .

Reply:
Hmm...I'm not sure what comment you were referring to there.

Anyway, -hzero was added to resolve a NaN issue (https://github.com/E3SM-Project/E3SM/issues/5053) in the PR  https://github.com/E3SM-Project/E3SM/pull/5208. 
So, are you sure it's not needed?

Reply:
In fact, I wonder if -hipa0 should be more limited in scope (in Depends file) as that would stop all interprocedural analysis and optimization including inlining of small functions.

Reply:
Andrew left a comment in scream PR (prob was erased by more commits):
```
Note we don't want to use -hzero, as that zeros all stack-allocated arrays. This can lead to uncaught uninitialized-variable bugs.
```

Reply:
Hmm. In one sense, -hzero was trying to address the unitialized variable issue. I think that should be fixed at the source level for all components. Unfortunately, I think there are issues with Land or FATES IIRC. Anyway, removing -hzero can potentially cause NaN's to return in certain test cases. Someone has to verify that it won't impact say coupled B-case MMF runs if we remove it.